### PR TITLE
MS Sculpt Mobile refactor

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Removing the use of the `ASTAR` define in favour of revision folders for the Teensy 2++ and Pololu A-Star configs.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by milestogo** [comment]
patch should not be needed if we can finish up https://github.com/qmk/qmk_firmware/pull/14461

**Comment by fauxpark** [comment]
There are some changes here that aren't in that PR.

**Review by tzarc — APPROVED** [review]

File: keyboards/handwired/ms_sculpt_mobile/astar/config.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/astar/config.h:1]: License header?

File: keyboards/handwired/ms_sculpt_mobile/teensy2pp/config.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/ms_sculpt_mobile/teensy2pp/config.h:1]: Also, license header?

File: keyboards/handwired/ms_sculpt_mobile/astar/config.h (hunk 1 above)
Comment (fauxpark) [reply keyboards/handwired/ms_sculpt_mobile/astar/config.h:1]: None of the existing files have one.

**Pull request merged**